.github/workflows/ci-workflow.yml

```diff
@@ -1,6 +1,19 @@
 name: Node Agent CI
 
-on: [push, pull_request, workflow_dispatch]
+on: 
+  - push:
+      paths: &path_anchor
+        - '**.js'
+        - 'bin/**'
+        # If this changed, it means we changed something in the code.
```

Comment:
including package-lock means this story will never be fulfilled.  releases edit package-lock.json

Reply:
Ah, forgot that the release number is also there. That's a bit of a pickle.